gnucash/gnome/gnc-split-reg.c

```diff
@@ -412,22 +468,40 @@ gsr_create_table( GNCSplitReg *gsr )
     Account * account = gnc_ledger_display_leader(gsr->ledger);
     const GncGUID * guid = xaccAccountGetGUID(account);
     gchar guidstr[GUID_ENCODING_LENGTH+1];
-    gchar *state_section = NULL;
+    gchar *register_state_section = NULL;
+    gchar *default_state_section = NULL;
+    gchar *group = NULL;
     guid_to_string_buff(guid, guidstr);
-    state_section = g_strconcat (STATE_SECTION_REG_PREFIX, " ", guidstr, NULL);
+
+    register_state_section = g_strconcat (STATE_SECTION_REG_PREFIX, " ", guidstr, NULL);
 
     ENTER("gsr=%p", gsr);
 
+    sr = gnc_ledger_display_get_split_register (gsr->ledger);
+    default_state_section = gnc_split_reg_get_state_group (sr);
+
+    // if this is from a page recreate and no register state use those settings,
+    // register state is dropped at the end of function.
+    if (gsr->page_state_name && !g_key_file_has_group (gnc_state_get_current (), register_state_section))
+        group = g_strdup (gsr->page_state_name);
```

Comment:
I didn't do the exercise here but I suspect you can do similar simplifications by ensuring group is always treated as a `const gchar*`

Reply:
Possibly, will have a look.